tool/release/CreateNotes.kt

```diff
@@ -0,0 +1,216 @@
+                    patch = version2[2].toInt(),
+                    alpha = null
+                )
+            } else if (version1.size == 2) {
+                return Version(
+                    major = version2[0].toInt(),
+                    minor = version2[1].toInt(),
+                    patch = version2[2].toInt(),
+                    alpha = 1
+                )
+            } else {
+                return Version(
+                    major = version2[0].toInt(),
+                    minor = version2[1].toInt(),
+                    patch = version2[2].toInt(),
+                    alpha = version1[2].toInt()
+                )
+            }
+        }
+    }
+
+    override fun compareTo(other: Version): Int {
+        val major = major.compareTo(other.major)
+        if (major == 0) {
+            val minor = minor.compareTo(other.minor)
+            if (minor == 0) {
+                val patch = patch.compareTo(other.patch)
+                if (patch == 0) {
+                    if (alpha == null) {
+                        if (other.alpha == null) return 0
+                        else return 1
+                    } else {
+                        if (other.alpha != null) return alpha.compareTo(other.alpha)
+                        else return -1
+                    }
+                } else return patch
+            } else return minor
+        } else return major
+    }
+
+    override fun toString(): String {
+        return "$major.$minor.$patch" + (if (alpha != null) "-alpha-$alpha" else "")
+    }
+}
+
+data class CommitNote(val title: String, val desc: String, val type: CommitNoteType)
+
+enum class CommitNoteType { FEATURE, BUG, REFACTOR, OTHER }
+
+fun getLastRelease(org: String, repo: String, githubToken: String): Version? {
+    val response = NetHttpTransport()
+        .createRequestFactory()
+        .buildGetRequest(GenericUrl("https://api.github.com/repos/$org/$repo/releases"))
+        .setHeaders(HttpHeaders().setAuthorization("Token $githubToken").setAccept("application/vnd.github.v3+json"))
+        .execute()
+    val body = Json.parse(String(response.content.readBytes()))
+    val releases = body.asArray().map { e -> Version.parse(e.asObject().get("tag_name").asString()) }
+    return releases.maxOrNull()
+}
+
+fun getCommits(org: String, repo: String, from: Version?, to: String, githubToken: String): List<String> {
+    val from_ = "d67639340ebf55a76e1f8cbd0fd7194cd212da02" // from?.toString() ?: TODO("get first commit")
+    val response = NetHttpTransport()
+        .createRequestFactory()
+        .buildGetRequest(GenericUrl("https://api.github.com/repos/$org/$repo/compare/$from_...$to"))
+        .setHeaders(HttpHeaders().setAuthorization("Token $githubToken").setAccept("application/vnd.github.v3+json"))
+        .execute()
+    val body = Json.parse(String(response.content.readBytes()))
+    return body.asObject().get("commits").asArray().map { e -> e.asObject().get("sha").asString() }
```

Comment:
@lolski Kotlin allows but doesn't require naming lambda arguments:

```suggestion
    return body.asObject().get("commits").asArray().map { it.asObject().get("sha").asString() }
```

Reply:
Yes, I know of this, but it felt too magical and I am reluctant to use it.